carina-api/src/main/java/com/qaprosoft/carina/core/foundation/api/AbstractApiMethodV2.java

```diff
@@ -100,6 +131,50 @@ public Response callAPI() {
         return rs;
     }
 
+    /**
+     * Calls API expecting http status in response taken from @SuccessfulHttpStatus value
+     * 
+     * @return restassured Response object
+     */
+    public Response callAPIExpectSuccess() {
+        SuccessfulHttpStatus successfulHttpStatus = this.getClass().getAnnotation(SuccessfulHttpStatus.class);
+        if (successfulHttpStatus == null) {
+            throw new RuntimeException("To use this method please declare @SuccessfulHttpStatus for your AbstractApiMethod class");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://demo.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS00112)